File: js/magic8.js
/* Set a username
TODO: let variable be filled in by user
*/
let userName = 'Sarah';

(userName) ? console.log(`Hello, ${userName}!`) : console.log(`Hello, stranger!`);
  // Workflow: Set username. Username not empty? Hello, ${userName}! Else username = 'Stranger' then Hello, ${userName}! (set to lowercase because grammar)

/* Ask the 8 ball a question
TODO:
- let question be filled in by user
- switch statement for if there's a question or not. Refer Python version of code for wording of strings
*/
let userQuestion = 'Is this a question?';

console.log(`${userName} asked: "${userQuestion}"`);
// future: check for a question mark in string in userQuestion, adjust accordingly. Maybe RegEx to ensure only questions are asked?

/* RNG for 8 ball replies */
let randomNumber = Math.floor(Math.random() * 8);
console.log(randomNumber);

// 8 ball replies based on RNG above
let eightBall = '';
switch (randomNumber) {
  case 0:
    eightBall = 'It is certain';
    break;
  case 1:
    eightBall = 'It is decidedly so';
    break;
  case 2:
    eightBall = 'Reply hazy try again';
    break;
  case 3:
    eightBall = 'Cannot predict now';
    break;
  case 4:
    eightBall = 'Do not count on it';
    break;
  case 5:
    eightBall = 'My sources say no';
    break;
  case 6:
    eightBall = 'Outlook not so good';
    break;
  case 7:
    eightBall = 'Signs point to yes';
    break;
  default:
    console.log('Undefined answer.');
    break;
}

// Print answer
console.log(eightBall);

File: python/magic8.py
# Imports
import random

# Initial variables
name = "Joe"
question = "Am I alive?"
answer = ""

# TODO: have user type into name and question variables

# Core logic
random_number = random.randint(1, 11)

if random_number == 1:
  answer = "Yes - definitely."
elif random_number == 2:
  answer = "It is decidedly so."
elif random_number == 3:
  answer = "Without a doubt."
elif random_number == 4:
  answer = "Reply hazy, try again."
elif random_number == 5:
  answer = "Ask again later."
elif random_number == 6:
  answer = "Better not tell you now."
elif random_number == 7:
  answer = "My sources say no."
elif random_number == 8:
  answer = "Outlook not so good."
elif random_number == 9:
  answer = "Very doubtful."
elif random_number == 10:
  answer = "Possibly."
elif random_number == 11:
  answer = "Unsure, ask me later!"
else:
  answer = "Error"

# 8-Ball answers
if question == "" and name == "":
  print("Do you want to tear apart reality as we know it!? Ask me a question before shaking!")
elif question == "":
  print(name + ", do you *really* want to tear apart the space-time continuum? Ask me a question before shaking, I like having a reality to live in!")
elif name == "":
  print("Question: " + question)
else:
  print(name + " asks: " + question)
  print("Magic 8-Ball's answer: " + answer)
